src/include/optimizer/index_util.h

```diff
@@ -40,7 +41,7 @@ class IndexUtil {
     auto sort_col_size = prop->GetSortColumnSize();
     for (size_t idx = 0; idx < sort_col_size; idx++) {
       // TODO(wz2): Consider descending when catalog/index support
-      auto is_asc = prop->GetSortAscending(static_cast<int>(idx)) == optimizer::OrderByOrderingType::ASC;
+      auto is_asc = prop->GetSortAscending(static_cast<int>(idx)) == catalog::OrderByOrderingType::ASC;
```

Comment:
Good catch here. I don't think we want an ASSERT because we should just choose not to use the sort property in the index choice if we reach this point. However, I missed that the sort still exists even if it cannot be satisfied, and we should not push down the limit. `implementation_rules.cpp` has been updated accordingly, if you could take another pass at it.